src/translate.py

```diff
@@ -4582,38 +4665,28 @@ def global_decls(
                     # The size of the element type (not the size of the array type)
                     if element_type is None:
                         element_type = sym.type
+
+                    # If we don't know the type, we can't guess the array_dim
                     type_size = element_type.get_size_bytes()
-                    if not type_size:
```

Comment:
The changes in this section have almost effect on the output. The only change is that the sizes in the warning comments now use `format_int`, e.g. `type too large by 0x30`. (I verified this using the MM tests with the changes in `array_access_from_add` commented out.)

This change was for the refactor of `GlobalSymbol.potential_array_dim`, so `array_access_for_add` could use the same array heuristic. (I think also makes this logic a bit easier to follow?)